### PR TITLE
Fix filter pattern cheat sheet examples

### LINKED ISSUE #855: Fix miscellaneous errors in "Filter Pattern" section of GitHub Actions / Reference / Workflow syntax
### What article on docs.github.com is affected?

This pertains to https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions.  (I could have updated issue #825 but chose to make a new issue instead.)

### What part(s) of the article would you like to see updated?

This issue has to do with the [File pattern cheat sheet](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet) section of the article.

1.  The third sentence in the explanation of the meaning of `[]` is "For example, the range[0-9a-f] matches any digits or lowercase letter."  This, of course, is not true; it doesn't match "g" for example.  Either the range should be changed to `[0-9a-z]`, or the explanation should read "any digit or any of the lowercase letters `a` through `f`."  Note that in any case, strictly speaking, it should read "any digit", not "any digits", although I doubt anyone will be particularly confused by "any digits."
2. In the [Patterns to match branches and tags](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#patterns-to-match-branches-and-tags) and [Patterns to match file paths](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#patterns-to-match-file-paths) (sub)sections, all of the example matches appear to start with a hyphen.  For example, the very first branch-matching pattern shown is `feature/*`, and the example matches given appear to be `-feature/my-branch` and `-feature/your-branch`, which don't actually match the given pattern.  I'm guessing that the initial hyphen is meant to serve the function of a bullet point and not meant to be considered as part of the matched text, but this is rather confusing, especially since the matches also contain internal hyphens.  In some cases, hyphens appear at the start of the patterns themselves, but pattern examples such as `-!README.md` suggest that the hyphen is not meant as part of the pattern.

### PR DESCRIPTION
**Pull request by tomjschuster**
### Why:

fixes #855

In the [Filter pattern cheat sheet examples](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet) section of https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet:

1. Regex example for matching any number or lowercase letter is shown incorrectly as `[0-9a-f]` (any hexadecimal number)
2. Examples of patterns to match different paths are prepended with `-`, but because they are within table cells, they are shown as literal `-`'s instead of bullet points. Since many of the patterns actually include `-`, this is visually confusing.
dash-bullets

### What's being changed:

1. Correct the regex pattern for numbers/lowercase letters to `[0-9a-z]`
2. Remove the `-` before the pattern examples in the table cells (line breaks should be sufficient for separating the patterns)
two-linebreaks

### Check off the following:
- [x] All of the tests are passing.
- [x] I have reviewed my changes in staging. (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**)
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

File: content/actions/reference/workflow-syntax-for-github-actions.md [hunk 1]
```diff
@@ -1122,33 +1122,33 @@ For more information about branch, tag, and path filter syntax, see "[`on.<push|
...
-| -`main`<br/>-`releases/mona-the-octcat` | Matches the exact name of a branch or tag name. | -`main`<br/>-`releases/mona-the-octocat` |
-| `'*'` | Matches all branch and tag names that don't contain a slash (`/`). The `*` character is a special character in YAML. When you start a pattern with `*`, you must use quotes. | -`main`<br/>-`releases` |
-| `'**'` | Matches all branch and tag names. This is the default behavior when you don't use a `branches` or `tags` filter. | -`all/the/branches`<br/>-`every/tag` |
-| `'*feature'` | The `*` character is a special character in YAML. When you start a pattern with `*`, you must use quotes. | -`mona-feature`<br/>-`feature`<br/>-`ver-10-feature` |
-| `v2*` | Matches branch and tag names that start with `v2`. | -`v2`<br/>-`v2.0`<br/>-`v2.9` |
-| `v[12].[0-9]+.[0-9]+` | Matches all semantic versioning tags with major version 1 or 2 | -`v1.10.1`<br/>-`v2.0.0` |
+| `feature/*` | The `*` wildcard matches any character, but does not match slash (`/`). |  `feature/my-branch`<br/><br/>`feature/your-branch` |
```
Comment (tomjschuster) [inline content/actions/reference/workflow-syntax-for-github-actions.md:1125]: Adding an additional line break to make the separation clearer:
**1 line break:** :point_down: 
one-linebreak


**2 line breaks:** :point_down: 
two-linebreaks

**Review by SMAKSS — APPROVED** [review]

**Comment by janiceilene** [comment]
@tomjschuster Thanks so much for opening a PR! I'll get this triaged for review ⚡

**Comment by janiceilene** [comment]
Thanks for your patience! Our small team is working our way through reviewing all of the amazing contributions ✨

**Comment by tomjschuster** [comment]
Just commenting to keep this open. Hope everyone is enjoying their holidays.

**Review by skedwards88 — APPROVED** [review]
This looks great! Thank you for fixing these issues.

**Pull request merged**